readme.txt
=== SuperSlider-Menu ===
Contributors: Daiv Mowbray
Plugin URI: http://wp-superslider.com/
Tags: menu, categories, sidebar, widget, mootools
Requires at least: 2.6
Tested up to: 2.7
Stable tag: 0.4

This Animated menu plugin uses Javascript to dynamically expand or collapsable categories and posts.

== Description ==

This is your Animated menu plugin that uses Mootools 1.2 javascript to form a multi-level collapsable menu widget in your sidebar for your categories and posts. Automatic opener upon page change. Highly configurable, theme based design, css based animations.

It is parcially based off of the Collapsing Categories Plugin by Robert Felty.


== Installation ==

 - Unpackage contents to wp-content/plugins/ so that the files are in a superslider-menu directory. 
 - Activate the plugin, 
 - Configure plugin under > settings > SuperSlider-menu
 - Go to Design > Widgets , and drag over the superslider-menu Widget, configure widget.
 - (optional) move plugin sub folder plugin-data to your wp-content folder,
	under  > settings > SuperSlider-menu > option group, File Storage - Loading Options
	select "Load css from plugin-data folder, see side note. (Recommended)".


== OPTIONS AND CONFIGURATIONS ==

Available under > settings > SuperSlider-menu

	* theme files to use for menu.
	* menu accordion to work with click or mouseover
	* allow close all tabs
	* opacity transition
	* transition speed
	* transition type
	* tooltips on categories on or off
	* Vertical mouse tracer on or off
	* *  Mouse tracer location, right or left of menu (moved to css control)
	* Mouse tracer reaction speed.
	* to load or not Mootools.js
	* css files storage loaction

	*Advanced options
	* disable built in menu structure
	* user defined objects, holder, toggler, content, and toglink.


Available in the widget options pane:

	* Show post counts in Category links
	* Order of Categories
	* Order of posts
	* Show Rss link as text or image
	* Show pages as well as posts
	* Sort by category name or category id
	* Sort in ascending or descending order

== Themes ==

Create your own graphic and animation theme based on one of these provided

	* Available themes
		* default
		* blue
		* black
		* custom

== To Do ==

	* Enqueue javascript files
	* although sub categories are indented, sub sub cats are not yet.
	* extended post list needs a max height limit, with a hover scroll for the list
	* fix known bugs:
				- the "don't show posts" option dismounts the data list
				- show pages as well as categories fails
				- cookied togglers open fails in some usage situations.
				

== Report Bugs Request / Options / Functions ==

* for now please use the comments system at http://wp-superslider.com
	

== Frequently Asked Questions ==

=  The folding doesn't work at all =
	
	You may have a javascript conflict with jquery or mootools, which may be added to 
	your theme header by your theme or another plugin. Try disabling mootools
	in the file superslider.php

=  How do I change the style of the collapsing categories lists? =
  
  I recommend that you move the folder plugin-data to your wp-content folder
  if you already have a plugin-data folder there, just move the superslider folder.
  remember to change the css location option in the settings page for this plugin.
  Or edit directly: 
  wp-content/plugins/superslider-menu/plugin-data/superslider/ssmenu/custom.css.
  Alternatively, you can copy those rules into your WordPress themes, style file. 
  Then remember to change the css location option in the settings page for this plugin.
  

= The stylesheet doesn't seem to be having any effect? =
 
  Check this url in your browser:
  http://yourblogaddress/wp-content/plugins/superslider-menu/plugin-data/superslider/ssmenu/custom.css
  If you don't see a plaintext file with css style rules, there may be
  something wrong with your .htaccess file (mod_rewrite). If you don't know
  how to fix this, you can copy the style rules there into your themes style
  file.

= How do I use different graphics and symbols for collapsing and expanding? =

You can upload your own images to
http://yourblogaddress/wp-content/plugins/superslider-menu/plugin-data/superslider/ssmenu/img_custom

= I can't get including or excluding to work = 

Make sure you specify category names, not ids.

== Screenshots ==

1. a few expanded categories with blue theme, showing nested categories and posts
2. plugin settings screen 
3. widget options screen

== Demo ==

This plugin is in use here at 
	* <http://wp-superslider.com>
	*<http://portfolio.daivmowbray.com>

== CAVEAT ==

Currently this plugin relies on Javascript to expand and collapse the links.
If a user's browser doesn't support javascript the list of cats and posts will display normally.

== HISTORY ==

* 0.4 (2008/11/20)
	* Fixed bug - opacity on tooltips now works.
	* Moved option - Mouse tracer location, right or left of menu to css control.

* 0.3 (2008/11/19)
	* Fixed bug - activeID is NULL when on non menu page.

* 0.2 (2008/11/19)
	* Added mootools / css powered animated tooltips on categories
	* Added Option
		* tooltips on categories  on / off

* 0.1.6 (2008/11/11)

	* Added settings screen for plugin.
	* Added Options
		* theme files to use for menu.
		* menu accordion to work with click or mouseover
		* allow close all tabs
		* opacity transition
		* transition speed
		* transition type
		* Vertical mouse tracer on or off
		* Mouse tracer location, right or left of menu
		* Mouse tracer reaction speed.
		* to load or not Mootools.js
		* css files storage loaction

		*Advanced options
		* disable built in menu structure
		* user defined objects, holder, toggler, content, and toglink.
	* Added theme system for menu.
	* Rebuilt html rendering file superslider-menu-list.php
	* Moved superslider-menu-ui.php into folder admin.
	* Added Reload default options to settings screen
	* Added settings link to wordpress plugin page
	* Added Set your widget link to settings page


* 0.1.2beta (2008/10/24)

	* Added mouse tracer - follower and option to deactivate it.
	* Added clicked link memory to identify and deactivate the active page link.
	* Added transition to css class for the accordion toggler.
	* Added toggler open memory (via cookie) for page changes.
	* Moved all java script into file superslider_menu.js


* 0.1.0beta (2008/10/15)

    * Changed name from Collapsing categories to Superslider-menu
    * Changed author from Robert Felty to Daiv Mowbray
    * Switched javascript from jquery to Mootools 1.2
    * switched html from unordered list to data list
    * Changed fold down icon to css based rather than image.
    * remove options:
    			- fold down icon type
    			- to animate or not
    			- expanding to show Cats and posts or just cats.
    			- auto expand these categories
    			- Category name as link to cat or not.
    * other...

---------------------------------------------------------------------------

CHANGELOG
SuperSlider-Menu Changelog

* 0.4 (2008/11/20)
	* Fixed bug - opacity on tooltips now works.
	* Moved option - Mouse tracer location, right or left of menu to css control.

* 0.3 (2008/11/19)
	* Fixed bug - activeID is NULL when on non menu page.

* 0.2 (2008/11/19)
	* Added mootools / css powered animated tooltips on categories
	* Added Option
		* tooltips on categories  on / off

* 0.1.6 (2008/11/11)

	* Added settings screen for plugin.
	* Added Options
		* theme files to use for menu.
		* menu accordion to work with click or mouseover
		* allow close all tabs
		* opacity transition
		* transition speed
		* transition type
		* Vertical mouse tracer on or off
		* Mouse tracer location, right or left of menu
		* Mouse tracer reaction speed.
		* to load or not Mootools.js
		* css files storage loaction

		*Advanced options
		* disable built in menu structure
		* user defined objects, holder, toggler, content, and toglink.
	* Added theme system for menu.
	* Rebuilt html rendering file superslider-menu-list.php
	* Moved superslider-menu-ui.php into folder admin.
	* Added Reload default options to settings screen
	* Added settings link to wordpress plugin page
	* Added Set your widget link to settings page


* 0.1.2beta (2008/10/24)

	* Added mouse tracer - follower and option to deactivate it.
	* Added clicked link memory to identify and deactivate the active page link.
	* Added transition to css class for the accordion toggler.
	* Added toggler open memory (via cookie) for page changes.
	* Moved Accordion java script into file superslider_menu.js


* 0.1.0beta (2008/10/15)

    * Changed name from Collapsing categories to SuperSlider-menu
    * Changed author from Robert Felty to Daiv Mowbray
    * Switched javascript from jquery to Mootools 1.2
    * switched html from unordered list to data list
    * Changed fold down icon to css based rather than image.
    * removed options:
    			- fold down icon type
    			- to animate or not
    			- expanding to show Cats and posts or just cats.
    			- auto expand these categories
    			- Category name as link to cat or not.

**********************  FORK  *******************************

* 0.6.3 (2008/10/03)
    * Added option to animate expanding and collapsing
    * Added option to add rss feeds for each category

* 0.6.2 (2008/09/11)
    * Fixed display of expand and collapse symbols when using images
    * Improved font handling and styling of text symbols

* 0.6.1 (2008/09/01)
    * Improved styling so that collapsing and expanding symbols use a
      fixed-width font, but category names do not
    * When using the option to have category names trigger expansion, and not
      showing posts, categories with no subcategories now link to the category
    * Added option to use images instead of html for collapse/expand characters
    * +/- now uses UTF-8 encoding instead of html entities (may not work for
      pages not encoded in UTF-8
    
* 0.6 (2008/08/27)
    * Can have multiple instances of widgets, each with separate options
    * No longer works as non-widget
    * All options are stored in one database row
    * Added more sorting options
    * Added option to include or exclude certain categories
    * Added option to expand certain categories by default
    * Added option to have category names either link to category archive or to
      activate expanding and collapsing

* 0.5.10 (2008/08/20)
    * minor bug fix. Fixed option to optionally show pages

* 0.5.9 (2008/08/07)
    * minor bug fix - added space before category count
    * Added option to sort by category (term) order
    * Added option to sort by category (term) count (note that it sorts by the
      count of the parent category, so categories with many subcategories, but
      not many posts themselves will be out of order
    * Added option to sort posts within categories

* 0.5.8 (2008/06/15)
		* bug fix - category description now correctly appears in title attribute
		  if there is a description for a given category
    * implemented a few more changes to work towards internationalization

* 0.5.7 (2008/05/23)
    * fixed misnamed class in javascript (collapsArch -> collapsCat)
    * added font-family definition to css to make it monospace for +/- 
    * added another option with brackets around the +/-

* 0.5.6 (2008/05/23)
    * fixed bug such that subcategories would not display the expand and
      collapse icons
    * fixed bug that categories with subcategories that have posts, but do not
      have posts themselves will be displayed
    * Thanks to [Andy] (http://www.onkelandy.com/blog) for both of these bug
      notices

* 0.5.5 (2008/05/19)
    * fixed bug - html now validates when not displaying posts
    * new option - choose between arrows or +- for expanding and collapsing
    * tweaked exclude option to function better with collapsing archives

* 0.5.4
    * fixed bug - was using hard-coded wp_ prefix in one SQL query. 
      Now using $wpdb-> instead

* 0.5.3
    * count is now correct for all subcategories

* 0.5.2
    * Added option to exclude certain categories
    * Added option to sort categories by slug

* 0.5.1
    * options in widget seem to work now
    * removed duplicate entries due to tag + category

* 0.5
    * Added option to not show posts
    * Added option to change title in widget
    * Now is condensed into one plugin

* 0.4.4
    * using unicode number codes in css stylesheet
    * fixed bug with duplicate entries in subcategories

* 0.4.3
    * nicer list indenting
    * re-fixed permalink bug introduced sometime after version 0.3.5

* 0.4.2
    * fixed bug with extraneous <ul>

* 0.4.1
		* fixed bug with get_sub_cat definition problem in WP 2.5. Looks like it
		  had something to do with nested functions maybe

* 0.4
    * Verified to work with wordpress 2.5
    * Now has custom styling option through the collapsCat.css stylesheet
    * updated screenshots

superslider-menu.php
<?php
/*
Plugin Name: SuperSlider-Menu
Plugin URI: http://wp-superslider.com/superslider-menu
Description: Animated Navigation List uses Mootools 1.2 javascript to expand and collapse categories, show subcategories and posts that belong to the category. 
Author: Daiv Mowbray
Version: 0.4
Author URI: http://wp-superslider.com
Tags: animation, animated, sidebar, widget, categories, mootools 1.2, mootools, menu, slider

Copyright 2008

       SuperSlider-Menu is free software; you can redistribute it and/or
    modify it under the terms of the GNU General Public License as published by 
    the Free Software Foundation; either version 2 of the License, or (at your
    option) any later version.

    SuperSlider-Menu is distributed in the hope that it will be useful,
    but WITHOUT ANY WARRANTY; without even the implied warranty of
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
    GNU General Public License for more details.

    You should have received a copy of the GNU General Public License
    along with Collapsing Categories; if not, write to the Free Software
    Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
*/

if (!class_exists("ssMenu")) {
	class ssMenu {
		
		/**
		* @var string   The name of this class.
		*/
	var $my_name;
	
	
	// Pre-2.6 compatibility
	function set_menu_paths()
	{
		if ( !defined( 'WP_CONTENT_URL' ) )
			define( 'WP_CONTENT_URL', get_option( 'siteurl' ) . '/wp-content' );
		if ( !defined( 'WP_CONTENT_DIR' ) )
			define( 'WP_CONTENT_DIR', ABSPATH . 'wp-content' );
		if ( !defined( 'WP_PLUGIN_URL' ) )
			define( 'WP_PLUGIN_URL', WP_CONTENT_URL. '/plugins' );
		if ( !defined( 'WP_PLUGIN_DIR' ) )
			define( 'WP_PLUGIN_DIR', WP_CONTENT_DIR . '/plugins' );
		if ( !defined( 'WP_LANG_DIR') )
			define( 'WP_LANG_DIR', WP_CONTENT_DIR . '/languages' );

	}
		/**
		* PHP 4 Compatible Constructor
		*/
	function ssmenu(){$this->__construct();}
		
		/**
		* PHP 5 Constructor
		*/		
	function __construct(){
		
		self::superslider_menu();
	
	}// end construct
		
		/**
		 * language switcher
		 */
	function language_switcher(){
		global $ssm_domain, $ssMenu_is_setup;

  				if($ssMenu_is_setup) {
     				return;
  				} 
  			// define some language related variables
		$ssm_domain = 'superslider-menu';
  			$ss_menu_locale = get_locale();
		$ss_menu_mofile = WP_LANG_DIR."/superslider_menu-".$ss_menu_locale.".mo";
  				//load the language
  			load_plugin_textdomain($ssm_domain, $ss_menu_mofile);
  			$plugin_text_loaded = true; // language is loaded
	}// end language switcher
	
			/**
		* Retrieves the options from the database.
		* @return array
		*/			
	function set_default_admin_options() {
		$defaultAdminOptions = array(
				"load_moo" => "on",
				"css_load" => "default",
				"css_theme" => "default",//end header ops 
				"user_objects" => "off",
				"holder" => "#ssMenuHolder",
				"toggler" => " dt span.show_",
				"content" => " dd.showme_",
				"toglink" => " dt a",
				"add_mouse" => "on",
				"alwayshide" => "on",
				"opacity" => "on",
				"trans_time" => "1250",
				"trans_type" => "quad",//end Accord ops 
				"trans_typeinout" => "in:out",
				"tooltips" => "on",
				"nav_follow" => "on",
				"nav_followside" => "right",
				"nav_followspeed" => "700");//end Follower ops
		
		$defaultOptions = get_option($this->AdminOptionsName);
		if (!empty($defaultOptions)) {
			foreach ($defaultOptions as $key => $option) {
				$defaultAdminOptions[$key] = $option;
			}
		}
		update_option($this->AdminOptionsName, $defaultAdminOptions);
		return $defaultAdminOptions;
		
	}

		/**
		* Saves the admin options to the database.
		*/
	function save_default_menu_options(){
		update_option($this->AdminOptionsName, $this->defaultAdminOptions);
	}

		/**
		* Load admin options page
		*/
	function ssmenu_ui() {
		//language_switcher();

		include_once 'admin/superslider-menu-ui.php';
		
	}

		/**
		* Initialize the admin panel, Add the plugin options page, loading it in from superslider-menu-ui.php
		*/
	function ssmenu_setup_optionspage() 
	{
		//language_switcher();
		/*global $ssMenu;
		if (!isset($ssMenu)) {
			return;
		}*/
		if( function_exists('add_options_page') ) {
			if( current_user_can('manage_options') ) {
			add_options_page(__('SuperSlider Menu'),__('ss-Menu'), 8, 'superslider-menu', array(&$this, 'ssmenu_ui'));
			add_filter('plugin_action_links', array(&$this, 'filter_plugin_menu'), 10, 2 );
			//this should print plugin info into admin footer
			//add_action('in_admin_footer', array(&$this, 'ssmenu_admin_footer'), 9 );
			}					
		}
	}
	function ssmenu_admin_footer() 
	{
		$plugin_data = get_plugin_data( __FILE__ );
		printf('%1$s plugin | Version %2$s | by %3$s<br />', $plugin_data['Title'], $plugin_data['Version'], $plugin_data['Author']);
	}

		/**
		* Add link to options page from plugin list.
		*/
	function filter_plugin_menu($links, $file) {
				 static $this_plugin;
 					if( ! $this_plugin ) $this_plugin = plugin_basename(__FILE__);

				if( $file == $this_plugin ){
					$settings_link = '<a href="options-general.php?page=superslider-menu">'.__('Settings').'</a>';
 					array_unshift( $links, $settings_link ); //  before other links
			}
 				return $links;
	}
  
   			/**
   			 * plughin settings link for WP 2.7
   			 *add_action('plugin_action_links_' . plugin_basename(__FILE__), 'filter_plugin_actions');
			function filter_plugin_actions($links) {
  				$settings_link = '<a href="options-general.php?page=myplugin">'.__('Settings').'</a>';
				array_unshift( $links, $settings_link ); // before other links
			return $links;
			}
   			 */
	function ssmenu_add_javascript(){
		$js_path = WP_CONTENT_URL . '/plugins/'. plugin_basename(dirname(__FILE__)) . '/js/';
		
		$ssmOpOut = get_option($this->AdminOptionsName);

		$loadMoo = ($ssmOpOut['load_moo']);	
		$navFollow = ($ssmOpOut['nav_follow']);
		$tooltips = ($ssmOpOut['tooltips']);
		
		echo "\t<!-- The following js is part of the SuperSlider-menu plugin available at http://wp-superslider.com/ -->\n";
	if (!is_admin())
			{				
				if (function_exists('wp_enqueue_script')) {
					if ($loadMoo == 'on'){
					//echo $js_path;
						//echo 'not admin navFollow on';
						//wp_enqueue_scrip('moocore', $js_path.'mootools-1.2-core.js');		
						//wp_enqueue_scrip('moomore', $js_path.'mootools-1.2-more.js', array('moocore'));
						echo "\t".'<script src="'.$js_path.'mootools-1.2-core.js" type="text/javascript"></script> '."\n";
						echo "\t".'<script src="'.$js_path.'mootools-1.2-more.js" type="text/javascript"></script> '."\n";
					};
					
					if ($navFollow == 'on'){
						echo "\t".'<script src="'.$js_path.'nav_follow.js" type="text/javascript"></script> '."\n";
						//wp_enqueue_script('nav_follow', $js_path.'nav_follow.js', array('$libtitle'));
					};
					//wp_enqueue_scrip('superslider_menu', $js_path.'superslider_menu.js', array($libtitle));
					echo "\t".'<script src="'.$js_path.'superslider_menu.js" type="text/javascript"></script> '."\n";						
				}	
			}
	}
		/**
		* register and Add js and css script into head 
		*/
	function ssmenu_add_css(){
		$ssmOpOut = get_option($this->AdminOptionsName);
		$cssLoad = ($ssmOpOut['css_load']);
		$cssTheme = ($ssmOpOut['css_theme']);	
		
		$url = get_settings('siteurl');

		echo "\t<!-- The following css is part of the SuperSlider-menu plugin available at http://wp-superslider.com/ -->\n";

    		if ($cssLoad == 'default'){
    			$cssPath = WP_PLUGIN_URL.'/superslider-menu/plugin-data/superslider/ssmenu/'.$cssTheme.'.css';
    			echo "\t"."<link rel='stylesheet' rev='stylesheet' href='".$cssPath."' media='screen' />\n";
    		}elseif ($cssLoad == 'pluginData') {
    			$cssPath = WP_CONTENT_URL.'/plugin-data/superslider/ssmenu/'.$cssTheme.'.css';
    			echo "\t"."<link rel='stylesheet' rev='stylesheet' href='".$cssPath."' media='screen' />\n";
    		}elseif ($cssLoad == 'off') {
    			$cssPath = '';
    		}

    		
	}	// end function ssmenu_add_css
	
			/**
		* Write and load the accordion code 
		*/
	function ssmenu_set_head() {
		$ssmOpOut = get_option($this->AdminOptionsName);
		
		$holder = $ssmOpOut['holder'];
		$toggler = $ssmOpOut['toggler'];
		$content = $ssmOpOut['content'];
		$toglink = $ssmOpOut['toglink'];
		$addMouse = $ssmOpOut['add_mouse '];		// on or off, activate toggler on mouse over -on on mouseclick-off.
		$alwaysHide = $ssmOpOut['alwayshide'];		// If set to on, it will be possible to close all displayable elements. Otherwise, one will remain open at all time.
		$opacity = $ssmOpOut['opacity'];			// set transition of opacity to on or off, fades the slider in and out as it slides.
		
		$transTime = $ssmOpOut['trans_time'];		// set the speed of the transition (milliseconds). 	
		$transType = $ssmOpOut['trans_type'];	// set the transition type, available: 
		$transTypeInOut = $ssmOpOut['trans_typeinout'];
		$transType = $transType.':'.$transTypeInOut;
		
		$tooltips = $ssmOpOut['tooltips'];
		
		$follow = $ssmOpOut['nav_follow'];		// set to 'follow' for on, and 'nofollow' for off.
		$fside = $ssmOpOut['nav_followside'];
		$fspeed = $ssmOpOut['nav_followspeed'];
		$float = '33';	
		/**/
		if ($tooltips == 'on'){
			$addTooltips = "				
				var myTips = new Tips($$('.tool'), {
								className: 'tipClosed',
								showDelay : 350,
								hideDelay : 80,
								offsets: {'x': -290, 'y': 0},
								fixed: 'true',
								timeOut: 900,
								maxTitleChars: 50,
								maxOpacity: .9,
					initialize:function(){
						this.fxopen = new Fx.Morph(this.tip, {
								duration: 750,
								transition: Fx.Transitions.Sine.easeInOut});
						this.fxclose = new Fx.Morph(this.tip, {
								duration: 250,
								transition: Fx.Transitions.Sine.easeInOut});
					},
					onShow: function(tip) {
						tip.fade(.8);
						this.fxopen.start('.customTip');	
					},
					onHide: function(tip) {
						tip.fade(0);
						this.fxclose.start('.tipClosed');
					}
				});";
		}else{
			$addTooltips = '';
		} 
		$ssmOpOut = ' "'.$holder.'","'.$toggler.'","'.$content.'","'.$toglink.'","'.$addMouse.'","'.$alwaysHide.'","'.$opacity.'","'.$transTime.'","'.$transType.'","'.$follow.'","'.$fside.'","'.$fspeed.'","'.$float.'"';
		echo "\t"."<script type=\"text/javascript\">\n";
		echo "// <![CDATA[\n";

		echo "window.addEvent('domready', function() {
				superslidermenu(".$ssmOpOut.");
				".$addTooltips."
				});\n";
		echo "\t"."// ]]>\n</script>\n";

		}
		
		/**
		* 
		*/		
	function ssmenu_init() {
		//language_switcher();
  				//load default options into data base
  			$this->defaultAdminOptions = $this->set_default_admin_options();
			$this->set_menu_paths();
	}// end ssmenu_init
	

	
	function superslider_menu()
	{
		
		$this->myname = get_class($this);
		$this->AdminOptionsName = 'ssMenu_options';
		//$plugin_text_loaded = false; // language is not loaded
		register_activation_hook(__FILE__, array(&$this,'ssmenu_init')); //http://codex.wordpress.org/Function_Reference/register_activation_hook
		register_deactivation_hook( __FILE__, array(&$this,'options_deactivation')); //http://codex.wordpress.org/Function_Reference/register_deactivation_hook
				
		add_action('init', array(&$this,'ssmenu_init')); 
		add_action('admin_menu', array(&$this,'ssmenu_setup_optionspage')); // start the backside options page
		add_action('plugins_loaded',array(&$this,'setup_widgets'));	
		$ssMenu_is_setup = 'true';
	
	}
	
		/**
		* Removes user set options from data base upon deactivation
		*/
	function options_deactivation(){
		delete_option($this->AdminOptionsName);
		delete_option(ssMenu_widget_options);
	}
	
		/**
		* 
		*/
	function foldcats($number) 
	{

		ssm_list_categories($number);		
	}
	function setup_widgets() 
	{	
		
		add_action('wp_head', array(&$this,'ssmenu_add_css'));
		add_action('wp_print_scripts', array(&$this,'ssmenu_add_javascript')); //this loads the mootools scripts.
		add_action('wp_head', array(&$this,'ssmenu_set_head')); //this writes the menu script into head.
		include('superslider-menu-list.php');
		include('superslider-menu-widget.php');
	}

}	//end class
} //End if Class ssMenu

/**
*instantiate the class
*/	
if (class_exists('ssMenu')) {
	$ssMenu = new ssMenu();
}

?>